dist/uPlot.cjs.js

```diff
@@ -1217,10 +1207,21 @@ function uPlot(opts, data) {
 				toggleDOM(ip, onOff);
 			}
 
-			// reset scale if current serie is actually the min or max bound
-			// if the toggled serie was the lower bound then s.min == scaleMin (same for max)
-			// the test then must be `s.min <= scaleMin` (or `s.max >= scaleMax`).
-			if (s.min <= scaleMin || s.max >= scaleMax) 
+			// compute new scale min and max 
+			var scaleMin = inf;
+			var scaleMax = -inf;
+			series.forEach(function (cs) {
+				if (cs.show) {
+					scaleMin = min(scaleMin, cs.min);
+					scaleMax = max(scaleMax, cs.max);
+				}
+			});
+			// snap scale
+			var sc = scales[s.scale];
+			var minMax = sc.range(scaleMin, scaleMax);
+			
+			// compare previous scale to new one
+			if (sc.min !== minMax[0] || sc.max !== minMax[1]) 
 				{ resetScale(s.scale); }
```

Comment:
Here I recompute the scale bounds from the cache min/max in series including the snap. If the scaled changed it needs to be reset. There is probably a better way to do it but it doesn't work :)

Reply:
i'm a bit embarrassed that i didnt notice this myself  😬. but yes, that's an issue!

Reply:
maybe still keeping the raw min/max cache at the scale level will be good, to reduce a bunch of extra loops. we'll see where i end up.